dos/src/main.rs

```diff
@@ -211,232 +279,274 @@ fn run_dos(
+    }
+    res
+}
 
-    #[clap(long, help = "Allow contacting private ip addresses")]
-    allow_private_addr: bool,
+fn create_payers<T: 'static + BenchTpsClient + Send + Sync>(
+    valid_blockhash: bool,
+    size: usize,
+    client: &Option<Arc<T>>,
+) -> Vec<Option<Keypair>> {
+    // Assume that if we use valid blockhash, we also have a payer
+    if valid_blockhash {
+        // each payer is used to fund transaction
+        // transactions are built to be invalid so the the amount here is arbitrary
+        let funding_key = Keypair::new();
+        let funding_key = Arc::new(funding_key);
+        let res = generate_and_fund_keypairs(
+            client.as_ref().unwrap().clone(),
+            &funding_key,
+            size,
+            1_000_000,
+        )
+        .unwrap_or_else(|e| {
+            eprintln!("Error could not fund keys: {:?}", e);
+            exit(1);
+        });
+        res.into_iter().map(Some).collect()
+    } else {
+        std::iter::repeat_with(|| None).take(size).collect()
+    }
+}
 
-    #[clap(flatten)]
+fn run_dos_transactions<T: 'static + BenchTpsClient + Send + Sync>(
+    target: SocketAddr,
+    iterations: usize,
+    client: Option<Arc<T>>,
     transaction_params: TransactionParams,
-}
+) {
+    let socket = UdpSocket::bind("0.0.0.0:0").unwrap();
 
-#[derive(Args, Serialize, Deserialize, Debug, Default)]
-#[clap(rename_all = "kebab-case")]
-struct TransactionParams {
-    #[clap(
-        long,
-        default_value = "2",
-        help = "Number of signatures in transaction"
-    )]
-    num_signatures: usize,
-
-    #[clap(long, help = "Generate a valid blockhash for transaction")]
-    valid_blockhash: bool,
+    let num_signatures = transaction_params.num_signatures;
+    let valid_signatures = transaction_params.valid_signatures;
+    let valid_blockhash = transaction_params.valid_blockhash;
 
-    #[clap(long, help = "Generate valid signature(s) for transaction")]
-    valid_signatures: bool,
+    // Number of payers is the number of generating threads, for now it is 1
+    // Later, we will create a new payer for each thread since Keypair is not clonable
+    let payers: Vec<Option<Keypair>> = create_payers(valid_blockhash, 1, &client);
+    let payer = &payers[0];
 
-    #[clap(long, help = "Generate unique transactions")]
-    unique_transactions: bool,
+    let mut transaction_generator = TransactionGenerator::new(transaction_params);
 
-    #[clap(
-        long = "payer",
-        help = "Payer's keypair file to fund transactions [Optional]"
-    )]
-    payer_filename: Option<String>,
-}
+    // Generate n=1000 unique keypairs, which are used to create
+    // chunks of keypairs in case of valid_signatures option
+    // The number of chunks is described by binomial coefficient
+    // and hence this choice of n provides large enough number of permutations
+    let mut keypairs_flat: Vec<Keypair> = Vec::new();
+    if valid_signatures || valid_blockhash {
```

Comment:
This part will not work properly for the case of 1-1 payment because it will produce duplicated transactions if num_signature != 1. This options should be irrelevant for this case but it is used. Requires introducing another cmd parameter number of instructions I guess